sources/packages/backend/apps/api/src/services/audit/audit.service.ts

```diff
@@ -0,0 +1,91 @@
+import { LoggerService } from "@nestjs/common";
+import { InjectLogger } from "@sims/utilities/logger";
+import { AuditEvent } from "./audit-event.enum";
+import { AuthorizedParties } from "../../auth";
+
+const SIMS_AUDIT_EVENT_PREFIX = "SIMS Audit Event";
+
+export class AuditService {
+  /**
+   * Audit logging.
+   * @param clientIp client ip.
+   * @param tokenUserName token user name.
+   * @param event audit event.
+   * @param authorizedParty authorized party.
+   */
+  async audit(
+    clientIp: string | string[],
```

Comment:
What is the scenario where the audit method will receive a `string[]`?

Reply:
`request.headers["x-forwarded-for"]` can return a `string[]`.

Reply:
Yes because it deals with headers. Is there a possibility that `x-forwarded-for` be returned twice?
This method cares about receiving an IP address only and not an array of them.